### PR TITLE
Add accessor for remaining time to `VerificationRequest`.

### PR DESCRIPTION
**Pull request by richvdh**
Element-web gives a countdown, so we need to know the remaining time.

### CODE REVIEW

File: bindings/matrix-sdk-crypto-js/src/verification.rs [hunk 1]
```diff
@@ -906,6 +908,20 @@ impl VerificationRequest {
         self.inner.timed_out()
     }
 
+    /// The number of milliseconds remaining before this verification flow times
+    /// out.
+    ///
+    /// Returns zero if the time has already passed.
+    #[wasm_bindgen(js_name = "timeRemainingMillis")]
+    pub fn time_remaining_millis(&self) -> f64 {
```
Comment (poljar) [inline bindings/matrix-sdk-crypto-js/src/verification.rs:916]: I feel like this belongs into the crypto crate. Otherwise we're running the risk of the other bindings re-implementing.

**Review by poljar — CHANGES_REQUESTED** [review]
Let's just put the `time_remaining()` method into the crypto crate as well.

File: bindings/matrix-sdk-crypto-js/src/verification.rs [hunk 2]
```diff
@@ -1073,6 +1084,8 @@ impl VerificationRequest {
     }
 
     /// Transition from this verification request into a SAS verification flow.
+    ///
+    /// Returns `Promise<[Sas, RoomMessageRequest|ToDeviceRequest] | undefined>`
```
Comment (richvdh) [inline bindings/matrix-sdk-crypto-js/src/verification.rs:1088]: oh, this is an unrelated thing that was apparently sitting in my working copy. It's pretty trivial so I'm inclined to leave it in this PR.

File: bindings/matrix-sdk-crypto-js/src/verification.rs (hunk 1 above)
Comment (richvdh) [reply bindings/matrix-sdk-crypto-js/src/verification.rs:916]: done.

**Review by poljar — APPROVED** [review]
Thanks, looks good.

File: bindings/matrix-sdk-crypto-js/src/verification.rs (hunk 2 above)
Comment (Hywan) [reply bindings/matrix-sdk-crypto-js/src/verification.rs:1086]: Yup, agreed.

File: bindings/matrix-sdk-crypto-js/src/verification.rs [hunk 3]
```diff
@@ -906,6 +906,15 @@ impl VerificationRequest {
         self.inner.timed_out()
     }
 
+    /// The number of milliseconds remaining before this verification flow times
+    /// out.
+    ///
+    /// Returns zero if the time has already passed.
+    #[wasm_bindgen(js_name = "timeRemainingMillis")]
+    pub fn time_remaining_millis(&self) -> f64 {
+        self.inner.time_remaining().as_millis() as f64
```
Comment (Hywan) [inline bindings/matrix-sdk-crypto-js/src/verification.rs:915]: This cast makes me uncomfortable. We are casting a `u128` to a `f64`. Casting to floats seems to always be a bad idea :-p.

What about returning a `u64`, or even a `u32`? There is `TryFrom<u128>` implementations for `u64` and `u32` as far as I reckon:

```rust
pub fn time_remaining_millis(&self) -> u64 {
    // SAFETY: `unwrap`ing is safe because the remaining time is always
    // lower than 2^64, making this `unwrap` unreachable.
    self.inner.time_remaining().as_millis().try_into().expect("failed to cast `time_remaining` to `u64`")
}
```

Thoughts?
It's

**Review by Hywan — APPROVED** [review]
I approve this PR, but I would like you to remove the cast from `u128` to `f64` please :-).

File: bindings/matrix-sdk-crypto-js/src/verification.rs (hunk 3 above)
Comment (richvdh) [reply bindings/matrix-sdk-crypto-js/src/verification.rs:915]: Well, I felt the same initially, so I tried returning a u64, but it got turned into a BigInt, which would have meant doing annoying conversions in the application. It's possible I was doing something wrong, but fundamentally, for better or worse, it's idiomatic to represent numbers as floating point in Javascript.

Given that, I think it makes sense for the bindings to cast to float.

In this case (a) we know it's going to be in the range [0,600000] which an f64 is perfectly capable of representing without loss of precision; (b) even that *wasn't* the case, we'd end up losing precision rather than range, and we don't really care about the difference between (a long time) and (a long time minus a few milliseconds).

Comment (Hywan) [reply bindings/matrix-sdk-crypto-js/src/verification.rs:915]: And what about returning a `u32` :-p? It's not a `BigInt` in JavaScript right?

Otherwise, fine, let's go with the actual solution.

Comment (richvdh) [reply bindings/matrix-sdk-crypto-js/src/verification.rs:915]: I didn't try a u32 tbh, but one way or the other we *want* to cast to a float for Javascript; we can either go via a u32 with fiddly error handling and adds no benefit in range or precision; or go straight to f64 which just works.
